chore!: Change graceful shutdown username to graceful-shutdown-user (#573)
stackabletech/trino-operator: docs/modules/trino/pages/usage-guide/operations/graceful-shutdown.adoc

== Authorization requirements

WARNING: When you are not using OPA for authorization, the user `graceful-shutdown-user` is not allowed to gracefully shut down workers.
If you need graceful shutdown you need to use OPA or need to make sure `graceful-shutdown-user` is allowed to gracefully shut down workers (e.g. having you own authorizer or patching Trino).

In case you use OPA to authorize Trino requests, you need to make sure the user `graceful-shutdown-user` is authorized to trigger a graceful shutdown of the workers.

> When you are using rego rules provided by Stackable this permission is automatically granted.

Reviewer: suggestion:

```suggestion
If you use rules provided by Stackable, this permission is automatically granted.
```

> However, when you are writing your own rego rules, you can achieve this e.g. by adding the following rule, which grants `graceful-shutdown-user` the permissions to issue a graceful shutdown.

Reviewer: suggestion:

```suggestion
If you use your own custom rego rules, you can achieve this by adding the following rule to grant `graceful-shutdown-user` the permissions to issue a graceful shutdown.
```

[source,rego]
----
allow {
  input.action.operation == "WriteSystemInformation"
  input.context.identity.user == "graceful-shutdown-user"
}
----

In Trino policies the following statement is recommended:
[source,rego]
----
"system_information": [
  # This is needed for graceful shutdown
  {
    "user": "graceful-shutdown-user",
    "allow": ["read", "write"]
  },
],

> ----

Reviewer 2: This policy is already provided by the Stackable OPA rules, so I would not mention or recommend it.

Author: Removed in d65e4d97dda7c7d128ae260fc19633dbd856d20a.
I was mostly thinking of users of Trino's file based authorizer, but we probably should not encourage that

> In case the user `admin` does not have the permission to gracefully shut down a worker, the error message `curl: (22) The requested URL returned error: 403 Forbidden` will be shown in the worker log and the worker will shut down immediately.

Reviewer 2: ```suggestion
In case the user `graceful-shutdown-user` does not have the permission to gracefully shut down a worker, the error message `curl: (22) The requested URL returned error: 403 Forbidden` will be shown in the worker log and the worker will shut down immediately.
```
